src/UseCases/BookPayment/BookPaymentUseCase.php

```diff
@@ -36,9 +36,21 @@ public function bookPayment( int $paymentId, array $transactionData ): SuccessRe
 			throw new \RuntimeException( 'Tried to book an non-bookable payment' );
 		}
 
-		if ( $payment->isCompleted() && $payment instanceof AssociablePayment ) {
+		if ( $this->isFollowUpPayment( $payment ) ) {
```

Comment:
PhpStan not seeing the narrowed type because it was extracted into a method here bums me out.